src/wms/src/main/java/org/geoserver/wms/map/RenderedImageMapOutputFormat.java

```diff
@@ -1029,14 +1030,14 @@ private RenderedImage directRasterRender(
                 //
                 // Get the reader
                 //
-                final Feature feature =
-                        mapContent
-                                .layers()
-                                .get(0)
-                                .getFeatureSource()
-                                .getFeatures()
-                                .features()
-                                .next();
+                final Feature feature;
+                try (FeatureIterator<?> it =
```

Comment:
Better than the previous code, but could be even better:

```
final Feature feature = DataUtilities.first(mapContent.layers().get(0).getFeatureSource().getFeatures().features());
```

Reply:
done